File: C/01_calculator.c
#include <stdio.h>
#include <math.h>
float permutation(int a, int b);
float combination(int a, int b);
int main()
{
    float a, b;
    long int c;
    char op;
    int isValid = 1;
    printf("Enter the first number : ");
	scanf("%f", &a);
	printf("Enter the operator : ");
	scanf(" %c", &op);
	printf("Enter the first number : ");
	scanf("%f", &b);
	switch (op)
    {
	case '+':
		c = a + b;
        break;
	case '-':
		c = a - b;
        break;
	case '*':
		c = a * b;
        break;
	case '/':
		if (b != 0)
			c = a / b;
		else
        {
			printf("\nDivision by 0 not possible !\n");
			isValid = 0;
		}
        break;
	case '^':
		c = pow(a, b);
        break;
	case '%':
		c = fmod(a, b);
        break;
		case 'P':
			if(b>a)
			{
				printf("\nPermutation not possible !\n");
			isValid=0;
			}
			c=permutation(a,b);
			break;
		case 'C': if(b>a)
		           {
		            printf("\nCombination not possible ! \n");
			    isValid=0;
		           }
			c=combination(a,b);
	default:
		printf("\nInvalid Operator !\n");
		isValid = 0;
        break;
	}
	if (isValid) {
		printf("\n%f %c %f = %ld\n", a, op, b, c);
	}
    return 0;
}
long permutation(int a, int b)
{
    return factorial(a) / factorial(a-b);
}

long combination(int a, int b)
{
    return permutation(a,b) / factorial(b);
}

long factorial(int num)
{
    long  fact = 1;

    while(num > 0)
    {
        fact *= num;
        num--;
    }

    return fact;
}

File: Dart/01_calculator.dart
import 'dart:io';
import 'dart:math';

void main() {
  stdout.write("Enter the first number : ");
  double a = double.parse(stdin.readLineSync());
  stdout.write("Enter the operator : ");
  String op = stdin.readLineSync();
  stdout.write("Enter the second number : ");
  double b = double.parse(stdin.readLineSync());
  double c;
  bool isValid = true;
  switch (op) {
    case '+':
      c = a + b;
      break;
    case '-':
      c = a - b;
      break;
    case '*':
      c = a * b;
      break;
    case '/':
      {
        if (b == 0) {
          stdout.write("\ndivision by 0 is not possible !\n");
          isValid = false;
        }
        c = a / b;
      }
      break;
    case '^':
      c = pow(a, b);
      break;
    case '%':
      c = a % b;
      break;
    default:
      {
        stdout.write("\nInvalid operator !\n");
        isValid = false;
      }
      break;
  }
  if (isValid) stdout.write("\n${a} ${op} ${b} = ${c}\n");
}
